reliability/single-fit/rum/src/test/kotlin/com/datadog/android/rum/integration/RumConfigurationTest.kt

```diff
@@ -0,0 +1,291 @@
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.startResource(originalResourceUrl, resourceMethod, originalResourceUrl, emptyMap())
+        rumMonitor.stopResource(originalResourceUrl, statusCode, size, resourceKind, emptyMap())
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        StubEventsAssert.assertThat(eventsWritten)
+            .hasRumEvent(1) {
+                hasResourceUrl(mappedResourceUrl)
+            }
+    }
+
+    @RepeatedTest(16)
+    fun `M send mapped events W setErrorEventMapper()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @StringForgery originalErrorMessage: String,
+        @StringForgery mappedErrorMessage: String,
+        @Forgery errorSource: RumErrorSource,
+        @StringForgery mappedErrorFingerprint: String,
+        @Forgery throwable: Throwable
+    ) {
+        // Given
+        val fakeRumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .setErrorEventMapper {
+                it.apply {
+                    error.message = mappedErrorMessage
+                    error.fingerprint = mappedErrorFingerprint
+                }
+            }
+            .build()
+        Rum.enable(fakeRumConfiguration, stubSdkCore)
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.addError(originalErrorMessage, errorSource, throwable, emptyMap())
+
+        // Then
+        val eventsWritten = stubSdkCore.eventsWritten(Feature.RUM_FEATURE_NAME)
+        StubEventsAssert.assertThat(eventsWritten)
+            .hasRumEvent(1) {
+                hasErrorMessage(mappedErrorMessage)
+                hasErrorFingerprint(mappedErrorFingerprint)
+            }
+    }
+
+    @RepeatedTest(16)
+    fun `M send mapped events W setActionEventMapper()`(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String,
+        @Forgery actionType: RumActionType,
+        @StringForgery originalTargetName: String,
+        @StringForgery mappedTargetName: String
+    ) {
+        // Given
+        val fakeRumConfiguration = RumConfiguration.Builder(fakeApplicationId)
+            .trackNonFatalAnrs(false)
+            .setActionEventMapper {
+                it.apply {
+                    action.target?.name = mappedTargetName
+                }
+            }
+            .build()
+        Rum.enable(fakeRumConfiguration, stubSdkCore)
+        val rumMonitor = GlobalRumMonitor.get(stubSdkCore)
+
+        // When
+        rumMonitor.startView(viewKey, viewName, emptyMap())
+        rumMonitor.startAction(actionType, originalTargetName, emptyMap())
+        rumMonitor.stopAction(actionType, originalTargetName, emptyMap())
+        Thread.sleep(100)
```

Comment:
why this call is needed if we called `stopAction`?

Reply:
Same as in the other class, actions stay alive for 100 ms to attach themselves to Errors, Long Tasks or Resources.